src/nemo_evaluator/observability/model_traffic.py

```
@@ -75,6 +75,25 @@ def _token_value(value: Any) -> int | None:
 
 
+def _token_list(value: Any) -> list[int] | None:
+    if not isinstance(value, list):
+        return None
+    try:
+        return [int(item) for item in value]
+    except (TypeError, ValueError):
+        return None
+
+
+def _first_token_list(data: Any, *keys: str) -> list[int] | None:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `data`

(ANN401)